src/integration/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/SharedTableBuilderIt.java

```diff
@@ -0,0 +1,98 @@
+import com.salesforce.dynamodbv2.mt.context.MtAmazonDynamoDbContextProvider;
+import com.salesforce.dynamodbv2.mt.context.impl.MtAmazonDynamoDbContextProviderThreadLocalImpl;
+import com.salesforce.dynamodbv2.mt.mappers.sharedtable.SharedTableBuilder;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+class SharedTableBuilderIt {
+
+    /**
+     * These tests run against a remote instance of DynamoDB, due to limitations in the LocalDynamoDBClient
+     * which does not allow BillingMode PAY_PER_REQUEST to be enabled without specifying throughput
+     * (which differs from what the DynamoDB API allows).
+     */
+
+    static final Regions REGION = Regions.US_EAST_1;
+    private static final String ID_ATTR_NAME = "id";
+    private static final String INDEX_ID_ATTR_NAME = "indexId";
+
+    AmazonDynamoDB remoteDynamoDB = AmazonDynamoDBClientBuilder.standard()
+        .withCredentials(new EnvironmentVariableCredentialsProvider())
+            .withRegion(REGION).build();
+
+    private static final String TABLE_PREFIX = "oktodelete-testBillingMode.";
+    String tableName;
+    public static final MtAmazonDynamoDbContextProvider MT_CONTEXT =
+            new MtAmazonDynamoDbContextProviderThreadLocalImpl();
+
+    @BeforeEach
+    void beforeEach() {
+        tableName = new String(String.valueOf(System.currentTimeMillis()));
+    }
+
+    @Test
+    void testBillingModePayPerRequestWithCustomTableRequest() {
+
+        CreateTableRequest request = new CreateTableRequest()
+                .withTableName(tableName)
+                .withKeySchema(new KeySchemaElement(ID_ATTR_NAME, HASH))
+                .withAttributeDefinitions(
+                        new AttributeDefinition(ID_ATTR_NAME, S),
+                        new AttributeDefinition(INDEX_ID_ATTR_NAME, S))
+                .withGlobalSecondaryIndexes(new GlobalSecondaryIndex()
+                        .withIndexName("index")
+                        .withKeySchema(new KeySchemaElement(INDEX_ID_ATTR_NAME, HASH))
+                        .withProjection(new Projection().withProjectionType(ProjectionType.ALL))
+                );
+
+        SharedTableBuilder.builder()
+                .withDefaultProvisionedThroughput(0)
+                .withCreateTableRequests(request)
+                .withStreamsEnabled(false)
+                .withPrecreateTables(true)
+                .withTablePrefix(TABLE_PREFIX)
+                .withAmazonDynamoDb(remoteDynamoDB)
+                .withContext(MT_CONTEXT)
+                .build();
+
+        assertEquals(BillingMode.PAY_PER_REQUEST.toString(), remoteDynamoDB.describeTable(TABLE_PREFIX + tableName)
+                .getTable().getBillingModeSummary().getBillingMode());
+
+        remoteDynamoDB.deleteTable(TABLE_PREFIX + tableName);
+    }
+
+    @Test
+    @Disabled
+    void testBillingModePayPerRequestWithDefaults() {
```

Comment:
If this is testing the default builder settings then I wouldn't expect to see `.withDefaultProvisionedThroughput(0)`.  If it isn't, then the test should be renamed.

Reply:
It's testing that Pay Per Request billing mode gets set for buildDefaultCreateTableRequests.  Renaming to "testBillingModePayPerRequestIsSetForDefaultCreateTableRequests"